aztec/src/main/kotlin/org/wordpress/aztec/toolbar/AztecToolbar.kt

```diff
@@ -559,7 +560,7 @@ class AztecToolbar : FrameLayout, IAztecToolbar, OnMenuItemClickListener {
         }
 
         // if text is selected and action is styling - toggle the style
-        if (action.isStylingAction() && action != ToolbarAction.HEADING && action != ToolbarAction.LIST) {
+        if (action.isStylingAction() && action != ToolbarAction.HEADING && action != ToolbarAction.LIST && action != ToolbarAction.CODE) {
```

Comment:
I think that this part:
```
      if (!editor!!.isTextSelected() && action.actionType == ToolbarActionType.INLINE_STYLE) {
            val actions = getSelectedActions()
            val textFormats = ArrayList<ITextFormat>()

            actions.filter { it.isStylingAction() }
                    .forEach { textFormats.add(it.textFormats.first()) }
```
could be applied even for heading/list/code if they are currently selected. The `getSelectedActions` takes whatever styling actions are now selected and applies them to the text format. 

> It seems to me that HEADING, LIST, and now CODE should not be styling actions because they won't be applied directly, just after the submenu is clicked or some specific logic to decide which format to apply (in case of CODE button). So instead of adding exceptions to this if statement for every toolbar action with a submenu, why don't we just declare these actions as not styling.

I think that it might be better to not rely on the `isStylingAction` method here because you're right, basically we skip that step if we want to manually handle the change (whether it's through a menu or in case of code where we want to be able to apply either block or inline action based on circumstances)